src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -151,4 +162,47 @@ public String getAdvancedPhylogeneticVisualizationPage(@PathVariable Long submis
 		model.addAttribute("submission", submission);
 		return BASE + "visualizations/phylocanvas-metadata";
 	}
+
+	/**
+	 * Get the html page from the file name provided.
+	 *
+	 * @param submissionId {@link Long} identifier for an {@link AnalysisSubmission}
+	 * @param filename     The html file name
+	 * @param model        {@link Model}
+	 * @param locale       User's locale
+	 * @return {@link String} path to the page template.
+	 */
+	@RequestMapping("/{submissionId}/html-output")
+	public String getHtmlOutputForSubmission(@PathVariable Long submissionId, @RequestParam String filename,
+			Model model, Locale locale) {
+		AnalysisSubmission submission = analysisSubmissionService.read(submissionId);
+		Set<AnalysisOutputFile> files = submission.getAnalysis()
+				.getAnalysisOutputFiles();
+		AnalysisOutputFile outputFile = null;
+		String htmlExt = "html";
+		String htmlOutput = "";
+
+		for (AnalysisOutputFile file : files) {
+			if (file.getFile()
+					.toFile()
+					.getName()
+					.contains(filename) && FilenameUtils.getExtension(filename)
+					.equals(htmlExt)) {
+				outputFile = file;
+				break;
+			}
+		}
+
+		try {
+			htmlOutput = FileUtils.readFileToString(outputFile.getFile()
```

Comment:
An issue I can see with this is if the HTML file is really big.  I've seen some tools which may (for example) dump a pile of data in a javascript object for displaying a result on a page.  In a case like this loading everything into a string in memory could cause a problem.

Alternative: can we stream this output directly as a response?  I think you can do something like that but I can't find an example in our existing codebase.  I know you can get a handle on the request & response objects in the controllers and write to them similar to what we do in `FastqView`.  Can you check if that's possible for these purposes as well?

Reply:
I can for sure take a look into if we can do that! Thanks :)

Reply:
Updated in f9123a19284ed01987451861e3da27613ae05b86